File: frontend/test-account-distribution.js
// Script para testar o novo endpoint de account distribution
// Execute no console do navegador em http://localhost:8080

console.log('🔧 Setting updated admin token for account distribution test...');

// Novo token válido
const token = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJhZG1pbiIsInJvbGUiOiJhZG1pbiIsImV4cCI6MTc1MDQwMTE3MCwiaWF0IjoxNzUwMzk3NTcwLCJ0eXBlIjoiYWNjZXNzX3Rva2VuIn0.HWlE5HTT7hWO0MO3lEV7bOv8LnCwLJCKiGFfHIv6s_g";

// Limpar e definir token
localStorage.clear();
localStorage.setItem('access_token', token);

console.log('✅ Token set successfully');
console.log('🧪 Testing account distribution endpoint...');

// Testar diretamente a API de account distribution
fetch('/api/v1/dashboard/account-distribution?provider=Oracle&time_filter=30d', {
  headers: {
    'Authorization': `Bearer ${token}`,
    'Content-Type': 'application/json'
  }
})
.then(response => response.json())
.then(data => {
  console.log('✅ Account distribution API test:', data);
  if (data.success && data.data.length > 0) {
    console.log(`📊 Found ${data.data.length} accounts for Oracle:`, data.data);
  }
})
.catch(error => {
  console.error('❌ Account distribution API test failed:', error);
});

console.log('🔄 Reloading page to test frontend integration...');
setTimeout(() => window.location.reload(), 2000);

File: backend/test_account_distribution.py
#!/usr/bin/env python3
"""
Script para testar o endpoint account-distribution existente
"""

import requests
import json

# Token válido
TOKEN = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJmaW5vcHNfYWRtaW4iLCJyb2xlIjoiZmlub3BzX2FkbWluIiwiZXhwIjoxNzUxOTQyMTQ4LCJpYXQiOjE3NTE4NTU3NDgsInR5cGUiOiJhY2Nlc3NfdG9rZW4ifQ.xM9WaQz1EbKedpVUzgicCqaiodPuBvxkiLYMOJjN9uc"

BASE_URL = "http://localhost:8000"
HEADERS = {
    "Authorization": f"Bearer {TOKEN}",
    "Content-Type": "application/json"
}

def test_account_distribution():
    """Testar endpoint account-distribution"""
    
    print("🧪 TESTANDO ENDPOINT ACCOUNT-DISTRIBUTION")
    print("=" * 60)
    
    tests = [
        {
            "name": "Oracle Cloud (nome exato)",
            "params": {
                "provider": "Oracle Cloud",
                "time_filter": "30d"
            }
        },
        {
            "name": "Oracle (nome normalizado)",
            "params": {
                "provider": "Oracle",
                "time_filter": "30d"
            }
        },
        {
            "name": "AWS (comparação)",
            "params": {
                "provider": "AWS",
                "time_filter": "30d"
            }
        }
    ]
    
    for test in tests:
        print(f"\n🔍 Teste: {test['name']}")
        
        try:
            response = requests.get(
                f"{BASE_URL}/api/v1/dashboard/account-distribution",
                headers=HEADERS,
                params=test["params"]
            )
            
            print(f"   Status: {response.status_code}")
            print(f"   URL: {response.url}")
            
            if response.status_code == 200:
                data = response.json()
                
                if 'data' in data:
                    accounts = data['data']
                    print(f"   ✅ Contas encontradas: {len(accounts)}")
                    
                    total_percentage = 0
                    for account in accounts:
                        print(f"      - {account['billing_account_name']}")
                        print(f"        ID: {account['account_id']}")
                        print(f"        Custo: ${account['total_cost']:,.2f}")
                        print(f"        Percentual: {account['percentage']:.1f}%")
                        total_percentage += account['percentage']
                    
                    print(f"   📊 Total percentual: {total_percentage:.1f}%")
                else:
                    print(f"   📄 Resposta: {data}")
            else:
                print(f"   ❌ Erro: {response.text}")
                
        except Exception as e:
            print(f"   ❌ Exceção: {e}")

if __name__ == "__main__":
    test_account_distribution()
